src/databricks/sql/utils.py

```diff
@@ -470,3 +474,66 @@ def _create_arrow_array(t_col_value_wrapper, arrow_type):
             result[i] = None
 
     return pyarrow.array(result, type=arrow_type)
+
+class DbSqlType(Enum):
+    STRING = "STRING"
+    DATE = "DATE"
+    TIMESTAMP = "TIMESTAMP"
+    FLOAT = "FLOAT"
+    DECIMAL = "DECIMAL"
+    INTEGER = "INTEGER"
+    BIGINT = "BIGINT"
+    SMALLINT = "SMALLINT"
+    TINYINT = "TINYINT"
+    BOOLEAN = "BOOLEAN"
+    INTERVAL_MONTH = "INTERVAL MONTH"
+    INTERVAL_DAY = "INTERVAL DAY"
+
+class DbSqlParameter:
+    name: str
+    value: Any
+    type: DbSqlType
+    def __init__(self,name="",value=None,type=None):
+        self.name = name
+        self.value = value
+        self.type = type
+        
+        
+def named_parameters_to_dbsqlparams(parameters: Any):
+    dbsqlparams = []
+    if(type(parameters) is dict):
+        # Support legacy impl
+        for name,parameter in parameters.items():
+            dbsqlparams.append(DbSqlParameter(name=name,value=parameter))
+    else:
+        for parameter in parameters:
+            if(type(parameter) is DbSqlParameter):
+                dbsqlparams.append(parameter)
+            else:
+                dbsqlparams.append(DbSqlParameter(value=parameter))
```

Comment:
I presume this branch of the else condition for line 504 is the case where parameters are passed as a `list`. In which case it seems like the parameter isn't named. In keeping with the idea that a function should do one thing, does this belong in a separate function?

Reply:
Well, the parameter might still be named. What we want the users to do is to use DBSQLParameters to pass in named parameters. That way they can encapsulate type as well.

Reply:
but ya, not really a way to decouple these: we need to support both the legacy and the new style.

Reply:
Will need to speak with you privately about this. My original feedback still stands. Nothing about this function resists splitting it into separate, testable methods.